icekit/navigation/abstract_models.py

```diff
@@ -0,0 +1,80 @@
+from django.core.urlresolvers import reverse
+from django.db import models
+from django.utils.six import python_2_unicode_compatible
+from django.utils.translation import ugettext_lazy as _
+from django.utils.six import text_type
+from fluent_contents.models import PlaceholderField, ContentItem
+from icekit.fields import ICEkitURLField
+from icekit.plugins import descriptors
+
+
+@python_2_unicode_compatible
+class AbstractNavigation(models.Model):
+    name = models.CharField(max_length=255)
+    slug = models.SlugField()
+    pre_html = models.TextField(blank=True)
+    post_html = models.TextField(blank=True)
+    content = PlaceholderField('navigation_content')
+
+    class Meta:
+        abstract = True
+
+    def __str__(self):
+        return self.name
+
+    def get_active_items_for_request(self, request):
```

Comment:
and add a `set_request` that sets `self.request` so that this property can use it.